subsys/usb/usb_c/usbc_pe_common_internal.h

```diff
@@ -22,6 +22,102 @@ enum pe_error {
 	ERR_XMIT,
 };
 
+/**
+ * @brief Policy Engine Layer States
+ */
+enum usbc_pe_state {
+	/** PE_SNK_Startup */
+	PE_SNK_STARTUP,
+	/** PE_SNK_Discovery */
+	PE_SNK_DISCOVERY,
+	/** PE_SNK_Wait_for_Capabilities */
+	PE_SNK_WAIT_FOR_CAPABILITIES,
+	/** PE_SNK_Evaluate_Capability */
+	PE_SNK_EVALUATE_CAPABILITY,
+	/** PE_SNK_Select_Capability */
+	PE_SNK_SELECT_CAPABILITY,
+	/** PE_SNK_Transition_Sink */
+	PE_SNK_TRANSITION_SINK,
+	/** PE_SNK_Ready */
+	PE_SNK_READY,
+	/** PE_SNK_Hard_Reset */
+	PE_SNK_HARD_RESET,
+	/** PE_SNK_Transition_to_default */
+	PE_SNK_TRANSITION_TO_DEFAULT,
+	/** PE_SNK_Give_Sink_Cap */
+	PE_SNK_GIVE_SINK_CAP,
+	/** PE_SNK_Get_Source_Cap */
+	PE_SNK_GET_SOURCE_CAP,
+	/**PE_Send_Soft_Reset */
+	PE_SEND_SOFT_RESET,
+	/** PE_Soft_Reset */
+	PE_SOFT_RESET,
+	/** PE_Send_Not_Supported */
+	PE_SEND_NOT_SUPPORTED,
+	/** PE_DRS_Evaluate_Swap */
+	PE_DRS_EVALUATE_SWAP,
+	/** PE_DRS_Send_Swap */
+	PE_DRS_SEND_SWAP,
+	/** PE_SNK_Chunk_Received */
+	PE_SNK_CHUNK_RECEIVED,
+
+	/** PE_Suspend. Not part of the PD specification. */
+	PE_SUSPEND,
+};
+
+/**
+ * @brief Policy Engine Layer Flags
+ */
+enum pe_flags {
```

Comment:
Note - in `struct policy_engine` defined in usbc_pe.h you declare flags as `atomic_t flags`.  This is usually limited to 32-bits.

You should either assert that `enum pe_flags` fits in 32-bits, or you need to update `struct policy_engine`:

```
struct policy_engine {
  ATOMIC_DEFINE(flags, PE_FLAGS_COUNT);
```

Zephyr's atomic support automatically supports more than 32-bit when using ATOMIC_DEFINE.

Reply:
Done